elements/ip/ipfilter.cc

```diff
@@ -1208,12 +1208,40 @@ IPFilter::Parser::parse_test(int pos, bool negated)
 }
 
 void
+merge_programs(Classification::Wordwise::Program &p1, Classification::Wordwise::Program &p2)
+{
+    int step_offset = p1.ninsn();
+    int jump_target = step_offset;
+    if (p1.ninsn() == 0) {
+        // p1 program is empty.
+        // If it has a valid target, we always jump to it. p2 is ignored.
+        // If target is invalid, proceed as normal. Resulting tree == p2.
+        int target = -p1.output_everything();
+
+        //check if target is valid
+        if (target != Classification::j_never + 1) {
+            return;
+        }
+    }
+    if (p2.ninsn() == 0) {
+        jump_target = -p2.output_everything();
+    }
+
+    p1.redirect_unfinished_insn_tree(jump_target);
+    p2.offset_insn_tree(step_offset);
+
+    for (int i = 0; i < p2.ninsn(); i++) {
+       p1.add_raw_insn(p2.insn(i));
+    }
+}
```

Comment:
I think this misses the case that both programs have 0 insns, but the first goes to an invalid target & the second goes to a valid one